src/HealthGPS.Console/csvparser.cpp

```diff
@@ -95,85 +97,86 @@ create_fields_index_mapping(const std::vector<std::string> &column_names,
     return mapping;
 }
 
-} // namespace detail
-
-bool load_datatable_from_csv(hc::DataTable &out_table, std::string full_filename,
-                             std::map<std::string, std::string> columns, std::string delimiter) {
+bool load_datatable_from_csv(hgps::core::DataTable &out_table, const std::string &filename,
+                             const std::map<std::string, std::string> &columns,
+                             const std::string &delimiter) {
     MEASURE_FUNCTION();
     using namespace rapidcsv;
 
-    Document doc(full_filename, LabelParams(0, 0), SeparatorParams(delimiter.front()));
+    bool success = true;
+    Document doc{filename, LabelParams{}, SeparatorParams{delimiter.front()}};
 
     // Validate columns and create file columns map
-    auto mismatch = 0;
     auto headers = doc.GetColumnNames();
-    std::map<std::string, std::string, hc::case_insensitive::comparator> csv_cols;
-    for (auto &pair : columns) {
-        auto is_match = [&pair](const auto &str) {
-            return hc::case_insensitive::equals(pair.first, str);
+    std::map<std::string, std::string, hc::case_insensitive::comparator> csv_column_map;
+    for (const auto &[col_name, col_type] : columns) {
+        auto is_match = [&col_name](const auto &csv_col_name) {
```

Comment:
Eh?

Reply:
This seems to be a bug in clang, where it can't capture structured bindings in lambdas, even though C++20 allows it. https://stackoverflow.com/questions/46114214/lambda-implicit-capture-fails-with-variable-declared-from-structured-binding

Work to fix it here: https://reviews.llvm.org/D122768

For now I guess we don't use structured bindings here, unless we can work around that lambda.